combine.go

```diff
@@ -60,6 +65,14 @@ func (t timeList) Less(i, j int) bool { return t[i].Before(t[j]) }
 func (t timeList) Swap(i, j int)      { t[i], t[j] = t[j], t[i] }
 
 func (n *CombineNode) runCombine([]byte) error {
+	valueF := func() int64 {
+		n.cardinalityMu.RLock()
```

Comment:
I like the idea of naming it after the thing its locking. The reason I named it `cardinalityMu` was so that it would be universal across each of the nodes.